Cython/Compiler/ExprNodes.py

```diff
@@ -9497,7 +9491,10 @@ def generate_evaluation_code(self, code):
                 member = struct_scope.lookup_here(item.key.value)
                 assert member is not None, f"struct member {item.key.value} not found, error was not handled during coercion"
                 key_cname = member.cname
-                value_cname = item.value.result()
+                value = item.value
+                if isinstance(value, CoerceToPyTypeNode):
+                    value = value.arg
```

Comment:
The code generation phase is too late to unpack this. It needs to be done earlier in a transform. When we get to the code generation, the tree needs to be correct.